### ISSUE TITLE
InitConfigDir is not platform agnostic

### ISSUE THREAD

**Issue by turetske** [PelicanPlatform/pelican#1834]
In the case where the $HOME or configDir directory isn't set, InitConfigDir makes an assumption that it's running on Linux.

This is rare case that could occur on Windows, but hasn't yet, so is probably why it hasn't been caught.

The InitConfigDir should be made platform agnostic and the tests should be allowed to run on windows to ensure everything works. We may need a discussion on what fall back config directory should be when not on Linux (as the code uses "/etc/pelican", which is a Linux path).

Basically, we want to ensure that for pelican clients running on either windows or linux, if the ConfigDir isn't set and the user isn't root, we have a fallback location that the client attempts to use on both.

**Comment by bbockelm** [comment]
@turetske - can you drop links into where the code makes these assumptions?

In general, the CI tests are indeed run on Windows -- are there specific ones that are currently marked as Linux-only that should be extended to Windows as well?  If so, can you drop links so one can easily pick up the ticket?
